### PR TITLE
Change database schema and routes to allow for assigning form instanc…

### PR DESCRIPTION
**Pull request by kaiyangzheng**
## Description

- Change SignatureField schema by removing signerPosition and signerPositionId
- Change Signature schema by adding signerType, signerDepartment, and signerPosition
- Modify formInstance service create and find assigned to to support signerDepartment and signerPosition

## Motivation and Context

- We want users to assign departments, positions, or users to signatures on form instances

Closes [#179]

## How has this been tested?

- Create form instance and template on web app

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by kaiyangzheng** [comment]
Todo: test and fix frontend (yarn gen-client) to work wth new backend schema/routes

**Comment by kaiyangzheng** [comment]
AssigneeMap on FormInstance - broken because we have names for departments and positions and it is not just firstname + last name anymore

Todo forms are not aligned correctly

Find a way for avatars to display info/differentiate between users, positions, and departments

Might rewrite AvatarMap because the html/css is bad

CreateFormInstanceMutation call is placeholder as USER for now and we manually set signerDepartmentId and signerPositionId as null

**Review by ahweigl — COMMENTED** [review]
LGTM, couple small comments/questions but otherwise good 🚀

File: apps/web/src/components/AvatarMap.tsx [hunk 1]
```diff
@@ -8,78 +8,90 @@ import { AvatarMapProps } from './types';
...
 const AssigneeMap: React.FC<AvatarMapProps> = ({ assignees }) => {
   let previousSigned = true;
+
+  const getInitialsFromTitle = (title: string, signerType: string) => {
+    if (signerType === 'DEPARTMENT') {
+      return 'D';
+    } else if (signerType === 'POSITION') {
+      return 'P';
+    } else {
+      return title;
```
[the remark this answers is not in the document]
Comment (kaiyangzheng) [reply apps/web/src/components/AvatarMap.tsx:18]: The initials of the user can just be their name when we pass it into the Avatar component

File: apps/web/src/client/models/CreateSignatureDto.ts [hunk 2]
```diff
@@ -5,7 +5,9 @@
 
 export type CreateSignatureDto = {
     order: number;
-    assignedUserId: string;
-    signerPositionId: string;
+    assignedUserId: string | null;
+    signerPositionId: string | null;
+    signerDepartmentId: string | null;
+    signerType: Record<string, any>;
```
[the remark this answers is not in the document]
Comment (kaiyangzheng) [reply apps/web/src/client/models/CreateSignatureDto.ts:11]: yes, but for some reason the prisma type in the client is Record<string, any> which is weird

File: apps/server/src/form-instances/form-instances.service.ts [hunk 3]
```diff
@@ -7,81 +7,50 @@ import { CreateFormInstanceDto } from './dto/create-form-instance.dto';
...
     private formTemplateService: FormTemplatesService,
-    private positionService: PositionsService,
     private employeeService: EmployeesService,
     private postmarkService: PostmarkService,
   ) {}
 
   /**
    * Create a new form instance.
-   * @param createFormInstanceDto
+   * @param createFormInstanceDto createFormInstanceDto
```
[the remark this answers is not in the document]
Comment (kaiyangzheng) [reply apps/server/src/form-instances/form-instances.service.ts:31]: github copilot typo lol

File: apps/server/src/form-instances/form-instances.service.ts [hunk 4]
```diff
@@ -128,19 +97,30 @@ export class FormInstancesService {
   }
 
   async findAssignedTo(employeeId: string) {
```
[the remark this answers is not in the document]
Comment (kaiyangzheng) [reply apps/server/src/form-instances/form-instances.service.ts:99]: Yea I'll add them

**Review by ahweigl — APPROVED** [review]
🚀

**Pull request merged**